modules/3d/include/opencv2/3d/volume_settings.hpp

```diff
@@ -0,0 +1,55 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+
+#ifndef OPENCV_3D_VOLUME_SETTINGS_HPP
+#define OPENCV_3D_VOLUME_SETTINGS_HPP
+
+#include <opencv2/core/cvstd.hpp>
+
+namespace cv
+{
+
+class CV_EXPORTS_W VolumeSettings
+{
+public:
+    VolumeSettings();
+    ~VolumeSettings();
+
+
+    void  setWidth(int val);
+    int   getWidth() const;
+    void  setHeight(int val);
+    int   getHeight() const;
+    void  setDepthFactor(float val);
+    float getDepthFactor() const;
+    void  setVoxelSize(float val);
+    float getVoxelSize() const;
+    void  setTruncatedDistance(float val);
+    float getTruncatedDistance() const;
+    void  setMaxWeight(int val);
+    int   getMaxWeight() const;
+    void  setRaycastStepFactor(float val);
+    float getRaycastStepFactor() const;
+    void  setZFirstMemOrder(bool val);
+    bool  getZFirstMemOrder() const;
```

Comment:
Not all the parameters require their own set/get methods.
For example, `zFirstMemOrder`, `width`, `height` and more

Reply:
why?
it doesn't take up a lot of space, but makes this class more flexible

Reply:
Regarding the last idea, about different intrinsics for `integrate` and `raycast`:
you may add an argument `InputArray intrinsics` to both functions. If it's not empty, then a user-passed value is used. If it's empty, a class-provided value is used.
This would keep a good balance between typical and pro- use cases.

Reply:
As it was discussed earlier, we don't need `zFirstMemoryOrder` exposed